### PR TITLE
Add more checks

### PR DESCRIPTION
**Pull request by blva**
1. [Add response-required-property-added checker](https://github.com/Tufin/oasdiff/pull/298/commits/66cd37bf601338cbf5e9576c2670a6c80889a111)
2. [Add response-required-property-added tests](https://github.com/Tufin/oasdiff/pull/298/commits/deb937395c9b8b3d9de62c46e769e15b705a9377)
3. [Add response-media-type-added checker and tests](https://github.com/Tufin/oasdiff/pull/298/commits/8edda0a70254925736508ae3a0f33ff4f143330e)
4. [Add security checkers](https://github.com/Tufin/oasdiff/pull/298/commits/bcd4bb2a0bddf945faae29d27a8dab9fcf9eb95f)
5.[Add request param type changed to changelog](https://github.com/Tufin/oasdiff/pull/298/commits/bfde1526ebb65d2737ba4e684df666befe8b4926)
6. [Add check-response-optional-property-updated checker](https://github.com/Tufin/oasdiff/pull/298/commits/4a0eee0b856d9014cb09744dcffc9e44decbdc78)
7. [Add response and response property schema type update checker](https://github.com/Tufin/oasdiff/pull/298/commits/169e8ad77cd043fd740e879ef9472099c56361f1)

### CODE REVIEW

File: checker/check-response-required-property-updated_test.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+// 			Id:          "response-required-property-added",
+// 			Text:        "added the required property 'data/new' to the response with the '200' status",
+// 			Comment:     "",
+// 			Level:       checker.INFO,
+// 			Operation:   "POST",
+// 			Path:        "/api/v1.0/groups",
+// 			Source:      "../data/checker/response_required_property_added_revision.yaml",
+// 			OperationId: "createOneGroup",
+// 		}}, errs)
+// }
```
Comment (matt-condon) [inline checker/check-response-required-property-updated_test.go]: Can you remove this if it's unused?

Comment (blva) [reply checker/check-response-required-property-updated_test.go]: thanks! removed

**Review by matt-condon — APPROVED** [review]
LGTM

File: checker/check-response-optional-property-updated.go [hunk 2]
```diff
@@ -32,25 +32,52 @@ func ResponseOptionalPropertyRemovedCheck(diffReport *diff.Diff, operationsSourc
...
 				modifiedMediaTypes := responseDiff.ContentDiff.MediaTypeModified
 				for _, mediaTypeDiff := range modifiedMediaTypes {
+					comment := ""
 					CheckDeletedPropertiesDiff(
 						mediaTypeDiff.SchemaDiff,
 						func(propertyPath string, propertyName string, propertyItem *openapi3.Schema, parent *diff.SchemaDiff) {
+							level := WARN
 							if propertyItem.WriteOnly {
-								return
+								level = INFO
```
Comment (blva) [inline checker/check-response-optional-property-updated.go:41]: Here, I'm making sure write-only properties will also be captured in the changelog. I don't think it's needed to create a separate text for it so Im adding it as a comment. Let me know if it makes sense or if we prefer to add a new checker id "response-optional-write-only-property-removed".

cc @tibulca @reuvenharrison

File: checker/check-request-body-required-value-updated_test.go [hunk 3]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/tufin/oasdiff/diff"
 )
 
-// BC: changing request's body to required is breaking
+// CL: changing request's body to required is breaking
```
Comment (ciprian-tibulca) [inline checker/check-request-body-required-value-updated_test.go:11]: is it intentional to change this from BreakingChange scenario to ChangeLog scenario?

File: checker/check-request-parameters-type-changed_test.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tufin/oasdiff/checker"
+	"github.com/tufin/oasdiff/diff"
+)
+
+// CL: Changing request path parameter type
+func TestRequestPathParamTypeChanged(t *testing.T) {
```
Comment (ciprian-tibulca) [inline checker/check-request-parameters-type-changed_test.go:12]: should this be a breaking change scenario and not changelog?

**Review by ciprian-tibulca — APPROVED** [review]

File: checker/check-response-optional-property-updated.go (hunk 2 above)
Comment (ciprian-tibulca) [reply checker/check-response-optional-property-updated.go:41]: Looking good, returning the INFO level will filter out this entry when running -check-breaking but will keep it when running -changelog.
Regarding the comment vs id, my preference would be to have a separate `response-optional-write-only-property-removed` id. Easier to (programmatically) process an entry with a specific id, than to check a comment.

File: checker/check-request-body-required-value-updated_test.go (hunk 3 above)
Comment (blva) [reply checker/check-request-body-required-value-updated_test.go:11]: BC entries are tracked in checker_breaking.go and checker_not_breaking.go 
- currently using check-<checker>_test.go files to track all changelog entries that are captured there. 
- open to discuss a target approach to capturing those entries and open a PR for that specifically

File: checker/check-response-optional-property-updated.go (hunk 2 above)
Comment (blva) [reply checker/check-response-optional-property-updated.go:41]: Thanks! Added that.

File: checker/check-request-body-required-value-updated_test.go (hunk 3 above)
Comment (ciprian-tibulca) [reply checker/check-request-body-required-value-updated_test.go:11]: thanks for clarifications

**Comment by reuvenharrison** [comment]
I'm reviewing...

File: checker/check-api-security-updated.go [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+		return result
+	}
+
+	for _, addedSecurity := range diffReport.SecurityDiff.Added {
+		result = append(result, BackwardCompatibilityError{
+			Id:          APIGlobalSecurityAddedCheckId,
+			Level:       INFO,
+			Text:        fmt.Sprintf(config.i18n(APIGlobalSecurityAddedCheckId), ColorizedValue(addedSecurity)),
+			Operation:   "N/A",
+			Path:        "N/A",
```
Comment (reuvenharrison) [inline checker/check-api-security-updated.go]: At some point in the future, we should add another type of BackwardCompatibilityError for components but for now, let's standardize the way we report them.

Compare with: https://github.com/Tufin/oasdiff/blob/e93e1ba256b7edc80af8bdecd337d1101afe8b36/checker/check-components-schemas-removed.go#L19

File: checker/check-api-security-updated.go [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+		}
+		for operation, operationItem := range pathItem.OperationsDiff.Modified {
+
+			source := (*operationsSources)[operationItem.Revision]
+
+			if operationItem.SecurityDiff == nil {
+				continue
+			}
+
+			for _, addedSecurity := range operationItem.SecurityDiff.Added {
```
Comment (reuvenharrison) [inline checker/check-api-security-updated.go:102]: Perhaps reuse func checkGlobalSecurity for these checks?

File: checker/localizations/localizations.go [hunk 7]
```diff
@@ -14,13 +14,30 @@ import (
 var localizations = map[string]string{
 	"en.messages.added-required-request-body":                              "added required request body",
 	"en.messages.api-deprecated-sunset-parse":                              "api sunset date '%s' can't be parsed for deprecated API: %v",
+	"en.messages.api-global-security-added":                                "the security %s was added to the API",
```
Comment (reuvenharrison) [inline checker/localizations/localizations.go]: I think the message should read:
"the security scheme %s was added to the API"

File: checker/localizations/localizations.go [hunk 8]
```diff
@@ -14,13 +14,30 @@ import (
...
+	"en.messages.api-global-security-updated":                              "the security %s was updated from %s to %s",
 	"en.messages.api-operation-id-added":                                   "api operation id %s was added",
 	"en.messages.api-operation-id-removed":                                 "api operation id %s removed and replaced with %s",
 	"en.messages.api-path-removed-before-sunset":                           "api path removed before the sunset date %s",
 	"en.messages.api-path-removed-without-deprecation":                     "api path removed without deprecation",
 	"en.messages.api-removed-before-sunset":                                "api removed before the sunset date %s",
 	"en.messages.api-removed-without-deprecation":                          "api removed without deprecation",
 	"en.messages.api-schema-removed":                                       "removed the schema %s from openapi components",
+	"en.messages.api-security-added":                                       "the endpoint security %s was added to the API",
```
Comment (reuvenharrison) [inline checker/localizations/localizations.go]: Perhaps:
"security scheme %s was added to the endpoint"

File: checker/localizations/localizations.go [hunk 9]
```diff
@@ -14,13 +14,30 @@ import (
...
+	"en.messages.api-security-component-added":                             "the component security schema %s was added",
```
Comment (reuvenharrison) [inline checker/localizations/localizations.go]: I think this should be "scheme" rather than "schema"

File: checker/checker_not_breaking_test.go [hunk 10]
```diff
@@ -258,7 +258,22 @@ func TestBreaking_Servers(t *testing.T) {
...
+	d, osm, err := diff.GetWithOperationsSourcesMap(getConfig(), &s1, &s2)
+	require.NoError(t, err)
+	errs := checker.CheckBackwardCompatibility(checker.GetDefaultChecks(), d, osm)
+	for _, err := range errs {
+		require.Equal(t, checker.ERR, err.Level)
+	}
+	require.Empty(t, errs)
+}
+
+// BC: adding a tag is not breaking with "api-tag-removed" check
```
Comment (reuvenharrison) [inline checker/checker_not_breaking_test.go]: Can you please explain what you mean by this: 'with "api-tag-removed" check'?

Comment (blva) [reply checker/checker_not_breaking_test.go]: I think I overspecified this BC, it's not needed and it should be simply // BC: adding a tag is not breaking

File: checker/localizations/localizations.go (hunk 9 above)
Comment (blva) [reply checker/localizations/localizations.go]: Thanks! updated

File: checker/check-api-security-updated.go (hunk 5 above)
Comment (blva) [reply checker/check-api-security-updated.go]: Thanks, makes sense, updated.

**Pull request merged**